pika/channel.py

```diff
@@ -390,7 +390,7 @@ def basic_reject(self, delivery_tag, requeue=True):
         """
         if not self.is_open:
             raise exceptions.ChannelClosed()
-        if not isinstance(delivery_tag, int):
+        if not is_integer(delivery_tag):
```

Comment:
My point is that people can register any type to be an `Integral`, and so I decided to not risk it. `NoneType` was just an extreme example.